treescript/src/treescript/gecko/android_l10n.py

```diff
@@ -0,0 +1,206 @@
+        repo_path (str): the source directory
+        description (str): commit message description
+        is_import (bool): True if import action
+
+    Returns:
+        int: non-zero if there are any changes.
+
+    """
+    ignore_closed_tree = get_ignore_closed_tree(task)
+    if not ignore_closed_tree:
+        if not await check_treestatus(config, task):
+            log.info("Treestatus is closed; skipping android-l10n import.")
+            return 0
+
+    dontbuild = get_dontbuild(task)
+
+    if is_import:
+        task_info = get_android_l10n_import_info(task)
+    else:
+        task_info = get_android_l10n_merge_info(task)
+    from_repo_path = tempfile.mkdtemp()
+    try:
+        from_repo_url = task_info["from_repo_url"]
+        if is_import:
+            cmd = ["git", "clone", from_repo_url, from_repo_path]
+            subprocess.run(cmd, text=True, check=True)
+        else:
+            await vcs.checkout_repo(config, task, from_repo_url, from_repo_path)
+
+        for toml_info in task_info["toml_info"]:
+            toml_path = os.path.join(from_repo_path, toml_info["toml_path"])
+            if is_import:
+                l10n_files = get_android_l10n_files_toml(toml_path)
+                src_path = None
+                dest_path = os.path.join(repo_path, toml_info["dest_path"])
+            else:
+                l10n_files = get_android_l10n_files_toml(toml_path, from_repo_path)
+                src_path = from_repo_path
+                dest_path = os.path.join(repo_path, toml_info["toml_path"])
+            os.makedirs(dest_path, exist_ok=True)
+            copy_android_l10n_files(l10n_files, src_path, dest_path)
+            shutil.copy2(toml_path, dest_path)
+
+        message = build_commit_message(description, dontbuild=dontbuild, ignore_closed_tree=ignore_closed_tree)
+        await vcs.commit(config, repo_path, message)
+    finally:
+        shutil.rmtree(from_repo_path, ignore_errors=True)
+
+    changes = 1
+
+    return changes
+
+
+# android_l10n_import {{{1
+async def android_l10n_import(config, task, repo_path):
+    """Perform a android_l10n string import.
+
+    This function takes its inputs from its task.
+    It reads the specified toml files to determine a list of
+    locale resource files from the android_l10n repo and copies
+    those from android_l10n into the source repo (typically
+    autoland).
+
+    Args:
+        config (dict): the running config
+        task (dict): the running task
+        repo_path (str): the source directory
+
+    Returns:
+        int: non-zero if there are any changes.
+
+    """
+    log.info("Preparing to import android-l10n changes.")
+
+    description = "Import translations from android-l10n"
+    changes = await android_l10n_action(config, task, repo_path, description, True)
+
+    return changes
+
+
+# android_l10n_merge {{{1
+async def android_l10n_merge(config, task, repo_path):
+    """Perform a android_l10n string merge.
```

Comment:
s/merge/sync/ maybe?  That matches the current action's naming and arguably is closer to what is happening.

Reply:
Good idea. I also considered "uplift" but I think I like "sync" best.